sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -639,4 +639,169 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
         "expressionStr" -> "SUBSTRING(struct1.a, 0, 1)",
         "reason" -> "generation expression cannot contain non-default collated string type"))
   }
+
+  test("Aggregation of arrays built on collated strings") {
+    val tableName = "test_agg_arr_collated"
+    val simple = Seq(
+      // binary
+      ("utf8_binary", Seq("array('aaa')", "array('AAA')"),
```

Comment:
Can you use named case objects instead of tuples? Not sure about the others, but I can't read this :)